corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -1291,27 +1291,28 @@ class ConfigureTableReportForm(ConfigureListReportForm):
     @property
     def _report_charts(self):
 
-        def get_non_agged_columns():
-            return [c for c in self._report_columns if c['aggregation'] != UCR_AGG_SIMPLE]
+        non_agged_columns = [c for c in self._report_columns if c['aggregation'] != UCR_AGG_SIMPLE]
+        agged_columns = [c for c in self._report_columns if c['aggregation'] == UCR_AGG_SIMPLE]
 
-        def get_agged_columns():
-            return [c for c in self._report_columns if c['aggregation'] == UCR_AGG_SIMPLE]
-
-        if get_non_agged_columns():
+        if non_agged_columns:
             if self.cleaned_data['chart'] == "bar":
                 return [{
                     "type": "multibar",
-                    "x_axis_column": get_agged_columns()[0]['column_id'] if get_agged_columns() else '',
+                    "x_axis_column": agged_columns[0]['column_id'] if agged_columns else '',
                     # TODO: Possibly use more columns?
                     "y_axis_columns": [
-                        {"column_id": c["column_id"], "display": c["display"]} for c in get_non_agged_columns()
+                        {"column_id": c["column_id"], "display": c["display"]} for c in non_agged_columns
                     ],
                 }]
             elif self.cleaned_data['chart'] == "pie":
+                if not agged_columns or not non_agged_columns:
```

Comment:
you already checked for `non_agged_columns` above so this check here should not be needed?

Reply:
oops good call. https://github.com/dimagi/commcare-hq/pull/21773/commits/2c84823b1ba7d47d441d2654e1ee3bd29973bd25